modules/light-clients/08-wasm/types/validation.go

```diff
@@ -19,3 +28,31 @@ func ValidateWasmCode(code []byte) error {
 func MaxWasmByteSize() uint64 {
 	return maxWasmSize
 }
+
+// ValidateWasmCodeHash validates that the code hash is of the correct length
+func ValidateWasmCodeHash(codeHash []byte) error {
+	lenCodeHash := len(codeHash)
+	if lenCodeHash == 0 {
+		return errorsmod.Wrap(ErrInvalidCodeHash, "code hash cannot be empty")
+	}
+	if lenCodeHash != 32 { // sha256 output is 256 bits long
+		return errorsmod.Wrapf(ErrInvalidCodeHash, "expected length of 32 bytes, got %d", lenCodeHash)
+	}
+
+	return nil
+}
+
+// ValidateClientID validates the client identifier
+func ValidateClientID(clientID string) error {
+	isClientID := clienttypes.IsValidClientID(clientID)
+	if !isClientID {
+		return errorsmod.Wrapf(ErrInvalidWasmClientID, "invalid client identifier %s", clientID)
+	}
+
+	hasWasmPrefix := strings.HasPrefix(clientID, exported.Wasm)
+	if !hasWasmPrefix {
+		return errorsmod.Wrapf(ErrInvalidWasmClientID, "client identifier %s does not contain %s prefix", clientID, exported.Wasm)
+	}
```

Comment:
nit: think in these cases we can make do w/o vars assigned, directly plop the function call in the `if`.

Reply:
done